### PR TITLE
Draft an issue triaging process proposal

### PR DESCRIPTION
**Pull request by hy9be**
Creating a local PR to get comments.

(Moved to https://github.com/maplibre/maplibre/pull/81)

### CODE REVIEW

File: docs/community/Issue-Triaging.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+Triaging an issue is a multi-step process that is collaboratively performed by community. Triaging an issue usually takes around one business day but may take longer, for example, when the reviewer is not around. Goal of triaging is to provide you with a clear understanding of what will happen to your issue. For example, after your feature request was triaged you know whether we plan to tackle the issue or whether we'll wait to hear what the broader community thinks about this request.
+
+
+## Our Triaging Process
+
+The basic process of triaging an issue is
+
+|First Step     |Next Step            |Result     |
+|---            |---                  |---        |
+|TSC review ->  |Accepted, or         |
```
Comment (nyurik) [inline docs/community/Issue-Triaging.md:12]: Define `TSC` somewhere above - not clear for the readers if you jump straight into TLAs

File: docs/community/Issue-Triaging.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+|               |Community review ->  |Closed, or |
+|               |                     |Accepted   |
+
+Please note:
+1. TSC reviewers can accept an issue but cannot close an issue before it is being reviewed by the community (the "Community Review" step).
+2. We only accept PRs for issues that are accepted.
```
Comment (nyurik) [inline docs/community/Issue-Triaging.md:18]: (2) is too strict - there are often cases of simple fixes, etc. I would word this as "we are much less likely to accept PRs that have no associated accepted issues" (grammar TBD)

File: docs/community/Issue-Triaging.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+|Feature Area|Description|
+|---|---|
+|[`rendering`](https://github.com/maplibre/maplibre-gl-native/labels/xx) | Rendering |
+|[`shader`](https://github.com/maplibre/maplibre-gl-native/labels/xx) | Shader |
+|[`styling`](https://github.com/maplibre/maplibre-gl-native/labels/xx) | Styling |
+
+### Platform Labels
+
+Each issue must have a **platform** label.
```
Comment (nyurik) [inline docs/community/Issue-Triaging.md:62]: There was some discussion on splitting the monorepo into multiple ones (undecided). How would this work for multiple repos targeting different platforms / sets of platforms?

File: docs/community/Issue-Triaging.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+If issues are suitable for beginners we may add the `good-first-issue` label and we add code pointers that help beginners to get started with a PR.
+
+Sometime, we get issues that we can't or don't have the time to reproduce due to the complexity or time requirements of the setup but that we indeed suspect to be issues. We label those issues with `investigation-wanted`. What we are looking for is help in reproducing and analyzing the issue.
+
+
+## Community Review
+
+### Accepting issue
+Community members can "up-vote" an issue (adding a GitHub +1/"👍" reaction to the issue description). If an issue surpasses the 20 up-votes in 60 days, the issue will be accepted.
```
Comment (nyurik) [inline docs/community/Issue-Triaging.md:93]: i don't think a formal "20 up-votes in 60" is a good measure. Each repo/target may have very different number of interest parties, plus there could be a lot of "yei" votes without explanation and a single but very clearly explained "no" vote that should overpower it.  We had similar issues with Wikipedia, and usually direct voting like this don't work well. Instead, the community simply tries to get to a consensus by discussing it, and usually arrives to a solution agreed by everyone. If not, there is always steering committee or the board to decide highly conflicted topics (hopefully very few)

File: docs/community/Issue-Triaging.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+### Closing issue
+
+Community members can also close an issue after reviewing it, for the following reasons:
+
+|Reason|Label|
+|---|---|
+|It's a duplicate of another issue. | `*duplicate`|
+|What is described is the designed behavior. | `*as-designed`|
+|The issue is a developer question, should be direct to Github Discussion or our Slack channel| `*dev-question`|
```
Comment (nyurik) [inline docs/community/Issue-Triaging.md:103]: auto-converting all dev/user questions to discussions seem excessive tbh.  Plus issues are fairly easy to search, manually or via the searching engine, so might make sense to answer it first, tag it as a question, and only then close it.  Plus a question is usually an indication of a missing/bad documentation :)

**Review by nyurik — COMMENTED** [review]
I do like the overall approach, thanks!  I did add a few minor points inline. One aspect we should keep in mind is that we don't have that many issues for the native lib (compared to some giant projects), so we don't have to be too strict with issue organization. The main goals for issue management should be - enough to be useful, but not too strict as to become a big project of itself.

**Comment by wipfli** [comment]
I think there are a lot of good ideas here, thanks @hy9be for suggesting this process. Probably it would be best to move this pull request over to maplibre-gl-native's main repo and continue the discussion there to get broader feedback.

**Comment by hy9be** [comment]
> I think there are a lot of good ideas here, thanks @hy9be for suggesting this process. Probably it would be best to move this pull request over to maplibre-gl-native's main repo and continue the discussion there to get broader feedback.

Sounds good. I will make some updates according to the feedback I get so far, and open a PR in the ~~main repo~~ maplibre/maplibre repo.

**Pull request closed without merging**